lib/dry/system/container.rb

```diff
@@ -91,7 +91,7 @@ def strategies(value = nil)
           if value
             @strategies = value
           else
-            @strategies ||= Dry::AutoInject::Strategies
+            @strategies || Dry::AutoInject::Strategies
```

Comment:
```suggestion
            if instance_variable_defined?(:@strategies) && @strategies
              @strategies
            else
              Dry::AutoInject::Strategies
            end
```

This should fix your undefined ivar warnings, @davydovanton?

Reply:
What I would probably do, though, is flatten everything into an if/elsif/else

```ruby
if value
  @strategies = value
elsif instance_variable_defined?(:@strategies) && @strategies
  @strategies
else
  Dry::AutoInject::Strategies
end
```

Reply:
thanks for the suggestion! I selected the second approach (for skipping nesting condition) 💯